Description: Prefer a /usr/share directory named after the package if it
  exists.
Author: Chad Miller <chad.miller@canonical.com>,
  Joachim Bauch <bauch@struktur.de>

--- a/chrome/common/chrome_paths.cc
+++ b/chrome/common/chrome_paths.cc
@@ -60,6 +60,8 @@
 #else
     FILE_PATH_LITERAL("/usr/share/chromium/extensions");
 #endif  // defined(GOOGLE_CHROME_BUILD)
+const base::FilePath::CharType kFilepathSinglePrefExtensionsNew[] =
+    FILE_PATH_LITERAL("/usr/share/iridium-browser/extensions");
 
 // The path to the hint file that tells the pepper plugin loader
 // where it can find the latest component updated flash.
@@ -466,6 +468,9 @@
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/opt/chrome/policies"));
 #else
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/chromium/policies"));
+      if (!base::PathExists(cur)) {
+        cur = base::FilePath(FILE_PATH_LITERAL("/etc/iridium-browser/policies"));
+      }
 #endif
       break;
     }
@@ -497,6 +502,10 @@
 #endif
 #if defined(OS_LINUX)
     case chrome::DIR_STANDALONE_EXTERNAL_EXTENSIONS: {
+      if (access(kFilepathSinglePrefExtensionsNew, R_OK|X_OK) == 0) {
+        cur = base::FilePath(kFilepathSinglePrefExtensionsNew);
+        break;
+      }
       cur = base::FilePath(kFilepathSinglePrefExtensions);
       break;
     }
@@ -547,6 +556,10 @@
 #else
       cur = base::FilePath(FILE_PATH_LITERAL(
           "/etc/chromium/native-messaging-hosts"));
+      if (!base::PathExists(cur))
+        cur = base::FilePath(FILE_PATH_LITERAL(
+            "/etc/iridium-browser/native-messaging-hosts"));
+
 #endif
 #endif  // !defined(OS_MACOSX)
       break;
